src/central-server/admin-service/ui/src/components/systemSettings/SelectMemberDialog.vue

```diff
@@ -0,0 +1,229 @@
+                  <td class="checkbox-column">
+                    <div class="checkbox-wrap">
+                      <v-radio :key="member.id" :value="member"></v-radio>
+                    </div>
+                  </td>
+                  <td>{{ member.member_name }}</td>
+                  <td>{{ member.xroad_id.member_code }}</td>
+                  <td>{{ member.xroad_id.member_class }}</td>
+                  <td>{{ member.xroad_id.subsystem_code }}</td>
+                  <td>{{ member.xroad_id.instance_id }}</td>
+                  <td>{{ member.xroad_id.type }}</td>
+                </tr>
+              </tbody>
+            </template>
+
+            <XrdEmptyPlaceholderRow
+              :colspan="7"
+              :data="filteredMembers()"
+              :loading="loading"
+              :no-items-text="$t('noData.noResults')"
+            />
+          </table>
+        </v-radio-group>
+      </v-card-text>
+      <v-card-actions class="xrd-card-actions">
+        <v-spacer></v-spacer>
+
+        <xrd-button
+          class="button-margin"
+          outlined
+          data-test="cancel-button"
+          @click="cancel()"
+          >{{ $t('action.cancel') }}</xrd-button
+        >
+
+        <xrd-button
+          :disabled="!selectedMember"
+          data-test="select-button"
+          @click="select()"
+          >{{ $t('action.select') }}</xrd-button
+        >
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Client } from '@/openapi-types';
+import { mapActions, mapStores } from 'pinia';
+import { clientStore } from '@/store/modules/clients';
+import { notificationsStore } from '@/store/modules/notifications';
+
+export default Vue.extend({
+  name: 'SelectMemberDialog',
+  props: {
+    dialog: {
+      type: Boolean,
+      required: true,
+    },
+    clientType: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      loading: false,
+      selectableClients: [] as Client[] | undefined,
+      search: '',
+      selectedMember: undefined,
+    };
+  },
+  computed: {
+    ...mapStores(clientStore),
+  },
+  created() {
+    this.loading = true;
+    this.clientStore
+      .getByClientType(this.clientType)
+      .then((resp) => {
+        this.selectableClients = resp;
+      })
+      .catch((error) => {
+        this.showError(error);
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  },
+  methods: {
+    ...mapActions(notificationsStore, ['showError', 'showSuccess']),
+    filteredMembers() {
+      if (!this.search) {
+        return this.selectableClients;
+      }
+
+      const tempSearch = this.search.toString().toLowerCase().trim();
+      if (tempSearch === '') {
+        return this.selectableClients;
+      }
+
+      return this.selectableClients?.filter((member) => {
+        if (member.member_name?.toLowerCase().includes(tempSearch)) {
```

Comment:
Perhaps instead of the extensive `if` - `else if` block would be nicer like this:
```
return member.member_name?.toLowerCase().includes(tempSearch) ||
  member.xroad_id.member_code.toLowerCase().includes(tempSearch) ||
  member.xroad_id.member_class.toLowerCase().includes(tempSearch) ||
  member.xroad_id.subsystem_code?.toLowerCase().includes(tempSearch) ||
  member.xroad_id.instance_id.toLowerCase().includes(tempSearch) ||
  member.xroad_id.type.toLowerCase().includes(tempSearch);
```